src/network/requests.ts

```diff
@@ -7,6 +7,45 @@ import {
   User
 } from "../types";
 
+export const addMoviePreference = (
+  user: User,
+  externalMovieId: number,
+  preferenceType: PreferenceType
+): void => {
+  fetch(`/users/${user.id}/movie-preferences`, {
```

Comment:
This could be a good time to factor out usages of `fetch` into `fetchGET`, `fetchPOST`, etc so that we can remove all the header duplication